src/main/java/org/alfresco/repo/security/authority/script/ScriptAuthorityService.java

```diff
@@ -372,7 +373,21 @@ public ScriptGroup getGroupForFullAuthorityName(String fullAuthorityName)
 		// group not found.
 		return null;
 	}
-	
+
+    /**
+     * Verifies whether there are authorities starting with the given displayNameFilter
+     *
+     * @param displayNameFilter filter (startsWith / ignoreCase) for authority display name
+     * @return true if there is at least an authority starting with given displayNameFilter, false otherwise
+     */
+	public boolean authoritiesExist(String displayNameFilter)
+    {
+        PagingResults<String> authorities = authorityService
+            .getAuthorities(AuthorityType.GROUP, null, displayNameFilter, false, false,
+                new PagingRequest(0, 1, null));
+        return authorities != null && !authorities.getPage().isEmpty();
+    }
+

```

Comment:
Is this needed? we have the same method as private in SiteServiceImpl...

Reply:
Thanks for review. This method will be used by the `sites.post` webscript in alfresco-remote-api -> https://github.com/Alfresco/alfresco-remote-api/pull/609/files#diff-43ccf775091f54ebf7aad38dfe4631f1R26 (next PR to be reviewed - https://github.com/Alfresco/alfresco-remote-api/pull/609). The webscript is not using `SiteServiceImpl` directly, but through `ScriptSiteService`. It is also using `ScriptAuthorityService` for authority specific calls. Thus if I want to use authoritiesExist(displayNameFilter) method in `sites.post` it needs to be defined in one of the script services mentioned above and `ScriptAuthorityService` is more appropriate for that. Let me know if this makes sense or there is a better solution.